llvm/lib/Target/AIE/AIEInterBlockScheduling.h

```diff
@@ -314,10 +315,17 @@ class InterBlockScheduling {
   /// This represents the resource usage we assume for an unscheduled successor.
   std::optional<int> getBlockedResourceCap(MachineBasicBlock *BB) const;
 
-  /// Return the number of nops that must be inserted before this block.
-  /// This is typical for epilogue blocks, which supply the safety margin of
-  /// their loops.
-  int getNumEntryNops(const BlockState &BS) const;
+  /// Return the number of nops that must be inserted before the epilogue
+  /// of the loop represented by this block.
+  int getSafetyMargin(MachineBasicBlock *Loop,
```

Comment:
Nit: safety-margin is very generic, and up to this day, we mostly used it to represent the distance to `ExitSU`. Maybe rename to `getEpilogueSafetyMargin`?